src/sql/src/plan/statement/dml.rs

```diff
@@ -160,25 +166,128 @@ pub fn plan_read_then_write(
     }))
 }
 
+fn is_side_effecting(
+    scx: &StatementContext,
+    select: &SelectStatement<Aug>,
+) -> Result<Option<SideEffectingFunction>, PlanError> {
+    match &select {
+        SelectStatement {
+            query:
+                Query {
+                    ctes: CteBlock::Simple(ctes),
+                    body: SetExpr::Select(body),
+                    order_by,
+                    limit: None,
+                    offset: None,
+                },
+            as_of: None,
+        } if ctes.is_empty() && order_by.is_empty() => match std::ops::Deref::deref(body) {
+            Select {
+                distinct: None,
+                projection,
+                from,
+                selection: None,
+                group_by,
+                having: None,
+                options,
+            } if from.is_empty()
+                && group_by.is_empty()
+                && options.is_empty()
+                && projection.len() == 1 =>
+            {
+                match projection.as_slice() {
+                    [SelectItem::Expr {
+                        expr:
+                            Expr::Function(Function {
+                                name,
+                                args: FunctionArgs::Args { args, order_by },
+                                filter: None,
+                                over: None,
+                                distinct: false,
+                            }),
+                        alias: None,
+                    }] if order_by.is_empty() => {
+                        fn eval_expr<'a>(
```

Comment:
```suggestion
                        fn eval_function_argument<'a>(
```

(Because the choice of `CastContext` is specific for function arguments.)